### ISSUE TITLE
Cloning async generator segfaults

### ISSUE THREAD

**Issue by acrylic-origami** [facebook/hhvm#7802]
### HHVM Version

[Produced in 3v4l](https://3v4l.org/33EOB) from 3.12.14 – 3.19, and locally:

```
$ hhvm --version
HipHop VM 3.16.0-dev (rel)
Compiler: heads/master-0-g07c7a02cd850b51c85ab64692298a91eb1c3e0c5
Repo schema: a0d9f666dd3c0dc8a940e03788cabf7bed52809d
```

### Standalone code, or other way to reproduce the problem

```hack
<?hh
$gen = async {
    yield 1;
};
$gen_clone = clone $gen;
```

### Expected result

Not sure. Unlike the original PHP interpreter, HHVM allows you to clone generators. The clones [[re-]advance their way through the generator separately from the original](https://3v4l.org/1SWtc). I'm not sure if async generators were intended to be cloneable too or not.

### Actual result

```
Core dumped: Segmentation fault
Stack trace in /tmp/stacktrace.27322.log
Segmentation fault (core dumped)
```

**Comment by enumag** [comment]
@acrylic-origami The clones do NOT readvance their way through the generator separately. In your example you just cloned the generator before any of it's code was executed. If you clone the generator after calling next you'll see that $i will be 1 instead of 2.

**Comment by acrylic-origami** [comment]
Yes you're right, my bad. "Re"-advance was poor wording; the two copies seem to advance separately from the state the original was cloned. https://3v4l.org/Hm8JP

**Comment by fredemmott** [comment]
Updated the description to say this still applies on 3.22

**Comment by lexidor** [comment]
I am going over old issues on this repository, to see which ones apply to the current versions of hhvm.

This doesn't segfault on hhvm 4.56, but it generates a Fatal Error (non-catchable, `\E_ERROR`) instead.

_Fatal error: NativeDataInfoCopy is not implemented yet. in /path/to/file.hack on line 6_
